src/app/workflow/launch-third-party/multi-cloud-launch/multi-cloud-launch.component.html

```diff
@@ -0,0 +1,59 @@
+<button mat-raised-button [matMenuTriggerFor]="menu">{{ this.partner }}</button>
+<mat-menu #menu="matMenu" xPosition="before" yPosition="below">
+  <div class="mat-menu-content-formatting" fxLayout="column" fxLayoutAlign="center start">
+    <p>Please choose a well-known server to launch on, or specify your own.</p>
+    <mat-radio-group
+      fxLayout="column"
+      fxLayoutAlign="center start"
+      name="launchWithUrl"
+      [(ngModel)]="launchWithOption"
+      (click)="$event.stopPropagation()"
+    >
+      <mat-radio-button
+        *ngFor="let instance of cloudInstances | filterCloudInstances: languagePartner"
+        name="launchWithUrl"
+        value="{{ instance.url }}"
+        [checked]="this.defaultLaunchWith === instance.url"
+      >
+        <span class="radio-options"> {{ instance.displayName }} </span>
+      </mat-radio-button>
+      <ng-container *ngIf="user">
+        <!--     Uncomment when we want to get the users custom cloud instances-->
+        <!--      <mat-radio-button-->
+        <!--        *ngFor="let usersInstance of usersCloudInstances | filterCloudInstances: languagePartner"-->
+        <!--        name="launchWithUrl"-->
+        <!--        value="{{ usersInstance.url }}"-->
+        <!--        [checked]="this.defaultLaunchWith === usersInstance.url"-->
+        <!--      >-->
+        <!--        {{ usersInstance.displayName }} (Custom)-->
+        <!--      </mat-radio-button>-->
+        <mat-radio-button value="other">
+          <mat-form-field color="accent">
+            <input
+              matInput
+              placeholder="Specify a url..."
+              [(ngModel)]="customLaunchWithOption"
+              (ngModelChange)="launchWithOption = 'other'"
+              name="something"
+            />
+          </mat-form-field>
+        </mat-radio-button>
+      </ng-container>
+    </mat-radio-group>
+  </div>
+  <div class="mat-menu-content-formatting" fxLayout="row" fxLayoutAlign="end end">
+    <button mat-button class="launch-buttons" color="accent" (click)="closeMatMenu()">Cancel</button>
+    <a
+      mat-raised-button
+      class="launch-buttons"
+      target="_blank"
+      rel="noopener"
+      [attr.href]="launchWith.url + launchUrlAppend"
+      color="accent"
+      [disabled]="this._launchWithOption === 'other' && (_customLaunchWithOption == null || _customLaunchWithOption === '')"
```

Comment:
In JavaScript, you can just do `!_customLanuchWithOption` -- catches null, undefined, and empty string. Not sure that it's a best practice though in TypeScript.